lib/helper.c

```diff
@@ -252,17 +252,20 @@ int fuse_parse_cmdline_30(struct fuse_args *args,
 }
 
 /**
- * Compatibility ABI symbol for systems that do not support version symboling
+ * Compatibility symbol for systems that do not support version symboling
  */
 #if (defined(__UCLIBC__) || defined(__APPLE__))
 int fuse_parse_cmdline(struct fuse_args *args,
 		       struct fuse_cmdline_opts *opts)
 {
-	return fuse_parse_cmdline_30(args, out_opts);
+#if FUSE_USE_VERSION < FUSE_MAKE_VERSION(3, 12)
```

Comment:
This is evaluated at compile time of the fuse library. However, the FUSE_USE_VERSION macro is defined at compile time of the application that links to FUSE. So I think that this will not have the intended effect, applications will get the most recent API even if they declare wanting to use an older one.

Reply:
I haven't taken into account this situation to be honest. I've pointed another approach below.

Reply:
I might be blind, but what is the final difference between both approaches? For initial patch, it only affects ULIBC and APPLE - is not global glibc problem? And then in either way for ulibc and apple - there is no way to have ABI compatibility, as the libc does not support it?

Reply:
Ah, actually the header approach should work better, because it gets compiled into the calling code. Better use "static inline" please.

Reply:
Yes, but with the first approach once you've built the library you can't change the FUSE_USE_VERSION anymore, while in the second approach you can. It's about redirecting fuse_parse_cmdline() to fuse_parse_cmdline_30/312() but while building an application that is linking against libfuse. I think this is what @Nikratio means.
But I'm not an expert of libfuse and I've never used FUSE_USE_VERSION to be honest.

Reply:
> Ah, actually the header approach should work better, because it gets compiled into the calling code. Better use "static inline" please, to avoid creating the same symbol multiple times.

Ok, I go for it then. Thanks both for reviewing!